service/matching/version_sets.go

```diff
@@ -126,13 +126,18 @@ func updateImpl(timestamp hlc.Clock, existingData *persistence.VersioningData, r
 	targetedVersion := extractTargetedVersion(req)
 	findRes := findVersion(existingData, targetedVersion)
 	targetSetIdx, versionInSetIdx := findRes.setIdx, findRes.indexInSet
+	numExistingSets := len(existingData.GetVersionSets())
 	modifiedData := persistence.VersioningData{
-		VersionSets:            make([]*persistence.CompatibleVersionSet, len(existingData.GetVersionSets())),
+		VersionSets:            make([]*persistence.CompatibleVersionSet, numExistingSets),
 		DefaultUpdateTimestamp: existingData.GetDefaultUpdateTimestamp(),
 	}
 	copy(modifiedData.VersionSets, existingData.GetVersionSets())
 
 	if req.GetAddNewBuildIdInNewDefaultSet() != "" {
+		// Make the request idempotent
+		if numExistingSets > 0 && targetSetIdx == numExistingSets-1 && versionInSetIdx == 0 && len(existingData.VersionSets[numExistingSets-1].BuildIds) == 1 {
```

Comment:
This is kinda hard to read. Maybe break these out into named bindings just to make it easy to grok